lib/src/parser/reader.rs

```diff
@@ -91,6 +91,12 @@ impl<T: ReadSeek> Reader<T> {
             )));
         }
 
+        if db_file_header.database_state == jet::DbState::DirtyShutdown {
```

Comment:
As a library, `ese_parser` should avoid side effects, like writing to the console. It's enough for `ese_parser` to report to user/client code that the database is dirty and then for API documentation to note that all bets are off (for now) if the database is dirty.

So, this may be as much an issue for sidr as for ese_parser.

Reply:
Done.